test/e2e/framework/framework.go

```diff
@@ -251,9 +251,18 @@ func (f *Framework) BeforeEach() {
 
 	if TestContext.GatherKubeSystemResourceUsageData != "false" && TestContext.GatherKubeSystemResourceUsageData != "none" {
 		var err error
+		var nodeMode NodesSet
```

Comment:
The changes to allow monitoring dns instances are pretty useful.

Can you please extract those to a separate PR and send it out for review?

Those will be beneficial for everyone.

Reply:
> Can you please extract those to a separate PR and send it out for review?

Yes - I was planning to do this today.